experimental/auth-logic/provenance_build_wrapper_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authlogic contains logic and tests for interfacing with the
+// authorization logic compiler
+package authlogic
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/project-oak/transparent-release/slsa"
+)
+
+func (p provenanceBuildWrapper) Identify() Principal {
```

Comment:
Is it possible to provide a blanket implementation for this, this code keeps repeating. Perhaps there can be another interface type for getting the path to a provenance file, and reuse the rest of the implementation.

Reply:
I think there's not too much repeating here, it's just:
 - getting the provenance struct from `ParseProvenanceFile`
 - getting the name from the provenance struct
A difference is that in the `EmitStatement()` function it needs to keep using the provenance struct for other things, so a function that just returns the name doesn't help. A function that returns a pair including the provenance and possibly an error is exactly `slsa.ParseProvenanceFile` which already exists.

So I don't really see a useful refactor here.

(https://github.com/project-oak/transparent-release/issues/40) will remove the common error handling code from both though

Reply:
Maybe I was not clear, this entire code block for implementing `Identify()` is repeated in all `test` files, and the only difference is in getting the path to a provenance file. I was not suggesting to share code between this and `EmitStatement ()`.

Reply:
It's not repeated in all the `test` files, the way `Identify()` works is different for the different wrappers. The implementation of `Identify()` is _similar_ for `provence_build_wrapper_test.go` and `provenance_wrapper_test.go` because the names both depend on the name in the provenance file, but the one for `build` has `Builder` in the name whereas the other one does not. The other implementations for `Identify()` are different.

I can also just make the test check the unattributed statement and drop the `Identify()` here completely.

Reply:
hmmm... I thought I was seeing the same code in many places, but maybe I just got confused with several PRs :)

Reply:
That's understandable (and there is nonzero overlap between the two provenance-related wrappers as discussed!).

I will mark this as resolved for now since I think it is resolved, but feel free to unresolve if needed.